packages/esm-patient-chart-app/src/visit/visit-form/visit-form.test.tsx

```diff
@@ -233,6 +271,260 @@ describe('Visit Form', () => {
     });
   });
 
+  it('starts a new visit with attributes upon successful submission of the form', async () => {
+    const user = userEvent.setup();
+
+    renderVisitForm();
+
+    mockOpenmrsFetch.mockResolvedValue({});
+    mockedUseConfig.mockReturnValueOnce({
+      visitAttributeTypes: [
+        {
+          uuid: visitAttributes.punctuality.uuid,
+          required: true,
+          displayInThePatientBanner: true,
+        },
+        {
+          uuid: 'aac48226-d143-4274-80e0-264db4e368ee',
+          required: false,
+          displayInThePatientBanner: true,
+        },
+      ],
+    });
+
+    const saveButton = screen.getByRole('button', { name: /Start visit/i });
+
+    // Set visit type
+    await user.click(screen.getByLabelText(/Outpatient visit/i));
+
+    // Set location
+    const locationPicker = screen.getByRole('combobox', { name: /Select a location/i });
+    await userEvent.click(locationPicker);
+    await user.click(screen.getByText('Inpatient Ward'));
+
+    const punctualityPicker = screen.getByRole('combobox', { name: 'Punctuality (optional)' });
+    await user.selectOptions(punctualityPicker, 'On time');
+
+    const insuranceNumberInput = screen.getByRole('textbox', { name: 'Insurance Policy Number (optional)' });
+    await user.clear(insuranceNumberInput);
+    await user.type(insuranceNumberInput, '183299');
+
+    mockSaveVisit.mockReturnValue(
+      of({
+        status: 201,
+        data: {
+          uuid: visitUuid,
+          visitType: {
+            display: 'Facility Visit',
+          },
+        },
+      }),
+    );
+
+    await user.click(saveButton);
+
+    expect(mockSaveVisit).toHaveBeenCalledWith(
+      expect.objectContaining({
+        location: mockLocations[1].uuid,
+        patient: mockPatient.id,
+        visitType: 'some-uuid1',
+      }),
+      new AbortController(),
+    );
+
+    expect(mockOpenmrsFetch).toHaveBeenCalledWith(`/ws/rest/v1/visit/${visitUuid}/attribute`, {
+      method: 'POST',
+      headers: { 'Content-type': 'application/json' },
+      body: { attributeType: visitAttributes.punctuality.uuid, value: '66cdc0a1-aa19-4676-af51-80f66d78d9eb' },
```

Comment:
Please try to avoid using magic UUIDs. It makes it hard to debug tests. I removed most of them with 5aa9965.